src/domain/checkForAndPlaceOrder.ts

```diff
@@ -533,152 +533,155 @@ async function _pollLegacy(
+        // The conditional order has signalled that it should be polled again on a specific epoch.
+        if (!blockNumberOrEpoch) {
+          throw new Error(
+            `Expected blockNumberOrEpoch to be defined for ${selector}`
           );
-          return {
-            result: PollResultCode.TRY_NEXT_BLOCK,
-            reason: `${selector}: ${message}`,
-          };
-        case "POLL_TRY_AT_BLOCK":
-          // The conditional order has signalled that it should be polled again on a specific block.
-          if (!blockNumberOrEpoch) {
-            throw new Error(
-              `Expected blockNumberOrEpoch to be defined for ${selector}`
-            );
-          }
-          return {
-            result: PollResultCode.TRY_ON_BLOCK,
-            blockNumber: blockNumberOrEpoch,
-            reason: `PollTryAtBlock: ${message}`,
-          };
-        case "POLL_TRY_AT_EPOCH":
-          // The conditional order has signalled that it should be polled again on a specific epoch.
-          if (!blockNumberOrEpoch) {
-            throw new Error(
-              `Expected blockNumberOrEpoch to be defined for ${selector}`
-            );
-          }
-          return {
-            result: PollResultCode.TRY_AT_EPOCH,
-            epoch: blockNumberOrEpoch,
-            reason: `PollTryAtEpoch: ${message}`,
-          };
-        case "POLL_NEVER":
-          // The conditional order has signalled that it should never be polled again.
-          return {
-            result: PollResultCode.DONT_TRY_AGAIN,
-            reason: `PollNever: ${message}`,
-          };
-      }
-    } catch (err: any) {
-      log.error(`${logPrefix} Unexpected error`, err);
-      return {
-        result: PollResultCode.UNEXPECTED_ERROR,
-        reason:
-          "UnexpectedErrorName: Unexpected error" +
-          (err.message ? `: ${err.message}` : ""),
-        error: err,
-      };
+        }
+        return {
+          result: PollResultCode.TRY_AT_EPOCH,
+          epoch: blockNumberOrEpoch,
+          reason: `PollTryAtEpoch: ${message}`,
+        };
+      case "POLL_NEVER":
+        // The conditional order has signalled that it should never be polled again.
+        return {
+          result: PollResultCode.DONT_TRY_AGAIN,
+          reason: `PollNever: ${message}`,
+        };
     }
+  } finally {
+    // We use finally here to ensure that we always log the error, even if we hit an unexpected error
+    // Any errors thrown here can _ONLY_ come from non-compliant interfaces (ie. bad revert ABI encoding).
+    // We log the error, and return a DONT_TRY_AGAIN result.
+    // TODO: Add metrics to track this
+    log.debug(
+      `Contract returned non-interface compliant revert via getTradeableOrderWithSignature. Simulate: https://dashboard.tenderly.co/gp-v2/watch-tower-prod/simulator/new?network=${chainId}&contractAddress=${to}&rawFunctionInput=${data}`
+    );
+    return {
+      result: PollResultCode.DONT_TRY_AGAIN,
```

Comment:
because the `case` is not exaustive, we can still end up here. Is this inteded we don't try again for unknown errors?

Reply:
The switch is now exhaustive with the use of typescript enum's / types. Turns out I was missing one! :see_no_evil:

Reply:
With the new typescript enum / types for the custom errors, the switch statement is now exhaustive.